x/budget/keeper/grpc_query.go

```diff
@@ -65,3 +66,25 @@ func (k Querier) Budgets(c context.Context, req *types.QueryBudgetsRequest) (*ty
 
 	return &types.QueryBudgetsResponse{Budgets: budgets}, nil
 }
+
+// Addresses queries an address that can be used as source and destination is derived according to the given name, module name and address type.
+func (k Querier) Addresses(_ context.Context, req *types.QueryAddressesRequest) (*types.QueryAddressesResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	if req.Name == "" && req.ModuleName == "" {
+		return nil, fmt.Errorf("at least one input of name or module name is required")
```

Comment:
To make the error handling consistent throughout the queries, I suggest to handle with grpc status package error `status.Error`.